frontend/apps/reader/modules/readertypography.lua

```diff
@@ -0,0 +1,802 @@
+                })
+                self.text_lang_tag = lang_tag
+                self.ui.document:setTextMainLang(lang_tag)
+                self.ui:handleEvent(Event:new("UpdatePos"))
+            end,
+            hold_callback = function(touchmenu_instance)
+                UIManager:show(MultiConfirmBox:new{
+                    -- No real need for a way to remove default one, we can just
+                    -- toggle between setting a default OR a fallback (if a default
+                    -- one is set, no fallback will ever be used - if a fallback one
+                    -- is set, no default is wanted; so when we set one below, we
+                    -- remove the other).
+                    text = T( _("Would you like %1 to be used as the default (★) or fallback (�) typography language?\n\nDefault will always take precedence while fallback will only be used if the language of the book can't be automatically determined."), BD.wrap(lang_name)),
+                    choice1_text = _("Default"),
+                    choice1_callback = function()
+                        G_reader_settings:saveSetting("text_lang_default", lang_tag)
+                        G_reader_settings:delSetting("text_lang_fallback")
+                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                    end,
+                    choice2_text = C_("Hyphenation", "Fallback"),
+                    choice2_callback = function()
+                        G_reader_settings:saveSetting("text_lang_fallback", lang_tag)
+                        G_reader_settings:delSetting("text_lang_default")
+                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                    end,
+                })
+            end,
+            checked_func = function()
+                return self.text_lang_tag == lang_tag
+            end,
+        })
+    end
+
+    self.language_submenu.max_per_page = 5
+    table.insert(self.menu_table, {
+        text_func = function()
+            local lang_name = LANG_TAG_TO_LANG_NAME[self.text_lang_tag] or self.text_lang_tag
+            return T(_("Typography language: %1"), lang_name)
+        end,
+        sub_item_table = self.language_submenu,
+    })
+
+    table.insert(self.menu_table, {
+        text = _("Follow document embedded lang tags"),
+        callback = function()
+            self.text_lang_embedded_langs = not self.text_lang_embedded_langs
+            self.ui.document:setTextEmbeddedLangs(self.text_lang_embedded_langs)
+            self.ui:handleEvent(Event:new("UpdatePos"))
+        end,
+        hold_callback = function()
+            local text_lang_embedded_langs = G_reader_settings:nilOrTrue("text_lang_embedded_langs")
+            UIManager:show(MultiConfirmBox:new{
+                text = text_lang_embedded_langs and _("Would you like to enable or disable support for document embedded lang tags by default?\n\nThe current default (★) is enabled.")
```

Comment:
Ditto.

Reply:
`_("Honor embedded lang tags")`
`Would you like to honor or ignore embedded lang tags by default?` is fine by me, because `Ignore | Honor` buttons rhymes likes `Disable | Enable` and I like when things sing :)
Fine with you ?

Reply:
Is this stuff about honoring Frenglish? Or are we just very corteous? :-P

Reply:
Dunno, It's understandable in that sense in french, and @NiLuJe uses it quite often in his comments. What would be another word for that, less generic than "use" ?
(But in french, it's also used in a meaning a bit more further than corteous, rather meaning intercour-teou-se :) which is the only reason I'm a bit not fond of using it for our text :)

Reply:
Well, what we want is a positive antonym to "ignore".
And there's not many of them in the various thesaurus and dict on the web.
"Honorer" is the right one in french - didn't see among the english antonyms to "ignore"...
The only one I found that could work in english in our context is: Obey (or Follow, that I had initially).

Reply:
apply

Reply:
`Apply document's embedded lang tags` ? So so, dunno...
What about my initial "Follow" ?

(There's "honor" in there :) http://www.antonymsfor.com/disregard )

Reply:
I didn't say honor doesn't mean that, but I think in French the meaning of granting/complying with/agreeing to/abiding by/conforming to/obeying may be less formal than in English.

Apply isn't intended to be synonymous, but as a rewrite to avoid the problem for non-native non-French speakers.

Reply:
But does everything have to be positive? I know there's a UI/UX school of thought that says it should be, but if the default is to abide by the lang tags I don't think it's weird to have an unchecked "Ignore lang tags" setting.

Or even just to leave out any kind of verb completely. The checkmark is a verb too. ^_^

Reply:
Overall, we are on the "check = positive = enable feature" side of UI thoughts :) (we have a few "Avoid...") - and in this new menu, all checkmarks are this positive/enable way. Would feel strange to have a "Ignore lang tags", even more because it's a new feature.
Still not fond of not having a verb, for this feature that is new and not really obvious and where "use" or "apply" do not convey much - while "honor" does.

> to avoid the problem for less advanced non-native non-French speakers

If "honor" works for english and french, can't we just get away with a translator note? :)

Reply:
I don't really see a difference between honor and apply tbh. ;-)

Reply:
I "honor lang tags" by giving them the typograhy rules they request/deserve :)
I can't "apply lang tags", this has no sense or feels incomplete - lang tags are there, they say something - I can't _apply them_, I can _apply what they say_.
I could "apply typography rules according to lang tags", but that's a bit long.

Reply:
<kbd>![image](https://user-images.githubusercontent.com/24273478/79732032-64524b00-82f3-11ea-9e69-05f33b17c5e7.png)</kbd>

to/with ? Or just `[ ] Adjust to embedded lang tags`

Reply:
If you don't honor lang tags it kind of feels like you're breaking an agreement with the document to me. :-P

> I can't "apply lang tags", this has no sense or feels incomplete - lang tags are there, they say something - I can't apply them, I can apply what they say.

But how is that any different for honor? I don't see the implied "content of" part as a problem either way.

Reply:
> If you don't honor lang tags it kind of feels like you're breaking an agreement with the document to me. :-P

That's why it's the default - should I remove "embedded styles off" and style tweaks, so we don't allow KOReader to break any agreement ? :)
(noticing you used naturally "If you don't honor lang tags" :)

> I don't see the implied "content of" part as a problem either way.

Well, I do :/
I can honor the Queen - by implicitely following/honoring her orders.
Can i apply the Queen ? :)

Requesting @NiLuJe 's input to settle this :)

Reply:
> (noticing you used naturally "If you don't honor lang tags" :)

But I was making a point about the word honor. The sentence "if you don't use lang tags it feels like you're breaking an agreement" wouldn't make any sense.

If you don't use or apply lang tags for typography it's a neutral choice.
If you don't honor lang tags it might imply more.

> I can honor the Queen - by implicitely following/honoring her orders.

That's a very different meaning of the word. One even more evocative of what I'm calling slightly odd in this context. Now you're going as far as to imply disrespect to the author if you don't honor the lang tags. ;-)

Reply:
So, any alternative suggestions to "honor lang tags" (that you don't like) and "use/apply lang tags" (that I don't like) ? (so we can stop trying to convince each other :)

`Adapt/adjust/use/blah rules to/with/for embedded lang tags` (with the right grammar) ?
(no need for the longer word "typography", it's there in the 2 menu items above.

Reply:
Enable?

Reply:
Full wording please.

Reply:
Enable embedded lang tags
Enable embedded lang tag typography
Enable embedded typography language

Although really I'm still in favor of having the checkmark as the verb.

Reply:
Really, I'm still in favor of Honor :)
Again, `Enable/apply/use lang tags` doesn't make any sense to me.
And "Enable embedded typography language" is super ambiguous (_what, there is a typography language inside HTML ? no one told me, let's enable it!_)

Please go along trying to find something else, with the vocabulary we use in the other menu item, and trying to be explicite rather than cryptic (at least, Honor is poetic :)
`[verb] rules [preposition] embedded lang tags` feels the clearest to me, in the context of this menu.
And we'll see if the verb is replaceable by the checkmark.

Reply:
You apply/use/honor lang tags [while rendering the document]. If `apply lang tags` doesn't make sense, surely neither does `honor lang tags`. But there's nothing *to* honor. There's no promise, agreement, debt or duty. But other than that minor shift in semantics I see no difference between the three.

Reply:
I understand you don't. But I do :) and you're not convincing me.
So, let's clear our mind from these and try to work out an alternative...

Reply:
Set typography based on (embedded) lang tags

Reply:
I'll readily admit to quite liking honor, possibly a bit too much ;).

A quick wordnet check shows

`
     * S: (v) respect, honor, honour, abide by, observe (show respect
       towards) "honor your parents!"
     * S: (v) honor, honour (accept as pay) "we honor checks and drafts"
`

`Abide by` is definitely too formal, but `respect` could work?

Reply:
Respect is probably the precise word you guys are actually looking for in honor. (Which isn't quite the same thing as being fully on board with it, but the objections I mentioned don't apply.)

Reply:
<kbd>![image](https://user-images.githubusercontent.com/24273478/79770958-638ada80-832e-11ea-80c5-4a9b063656f2.png)</kbd>

Fine with Respect, but somehow, after all this discussion, I now wish we had "rules" in here. "Honoring or respecting lang tags" feels more more enough :)

<kbd>![image](https://user-images.githubusercontent.com/24273478/79770780-27577a00-832e-11ea-928d-b39135201438.png)</kbd>

I could go with that - but "Set typography" feels a bit too global, could mean that the other settings will no more apply.
Anything less radical than "set", like adjust/adapt/tweak ?

`[ ] Typography rules by embedded lang tags` ? (no verb :) 
`[ ] Typography rules driven by embedded lang tags` ?

Reply:
I want to say `Typography rules as per embedded lang tags` but that's a bit science slangy. "by" is short for something like "as specified by" and I'm not quite sure if it works by itself.

Reply:
@poire-z: I think we can get away without `rules` since in a two-option section between separators with the one right above explicitly saying  `Typography rules` ;).

I find the other suggestions way too wordy, an possibly not quite as clear (of which I prefer @Frenzie's, FWIW ;)).

Reply:
> my vote would be for @Frenzie

Which one ?

Reply:
> Which one ?

I meant as far as the wordy ones are concerned: `Typography rules as per embedded lang tags`.

But, to be clear, I much prefer

`Respect embedded lang tags` ;).

Reply:
I'm fine with all of these:
`Typography rules as per embedded lang tags`
`Rules as per embedded lang tags`
`As per embedded lang tags`
`Respect embedded lang tags`

So, please you both agree on one :)

(Really nothing could work with an initial verb like Adapt/adjust - to not have the feeling that if you check that box, and the document has no embedded lang tags, you will get no rule applied?)

Reply:
The respect one is fine by me. (I thought I already posted this. :-) )